pkgs/applications/window-managers/sway/wrapper.nix

```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
+, sway-unwrapped
+, swaybg
+, dbus
+, wrapGAppsHook
+, symlinkJoin
+, writeShellScriptBin
+, withBaseWrapper ? true
+, withGTKWrapper ? false
+, extraSessionCommands ? ""
+}:
+
+with lib;
+
+let
+  swayPackage = sway-unwrapped;
+
+  baseWrapper = writeShellScriptBin "sway" ''
+    set -o errexit
+
+    if [ ! "$_SWAY_WRAPPER_ALREADY_EXECUTED" ]; then
+      export _SWAY_WRAPPER_ALREADY_EXECUTED=1
+      ${extraSessionCommands}
+    fi
+
+    if [ "$DBUS_SESSION_BUS_ADDRESS" ]; then
+      export DBUS_SESSION_BUS_ADDRESS
+      exec ${swayPackage}/bin/sway "$@"
+    else
+      exec ${dbus}/bin/dbus-run-session ${swayPackage}/bin/sway "$@"
+    fi
+  '';
+
+  swayJoined = symlinkJoin {
+    name = "sway-base-wrapped";
+    paths = (optional withBaseWrapper [baseWrapper]) ++ [swayPackage];
+  };
+in
+symlinkJoin {
+  name = "sway";
+
+  paths = [swayJoined];
+
+  version = sway-unwrapped.version;
+
+  buildInputs = []
+    ++ (optional withGTKWrapper [wrapGAppsHook])
```

Comment:
I think this belongs into `nativeBuildInputs`, then we can also drop `postBuild` (IIRC).

Reply:
If I drop the `postBuild` the GTK Wrapper seems to miss some variables (not really sure why) and it revert to the previous behavior.

Reply:
Did you move `wrapGAppsHook` to `nativeBuildInputs`?

Reply:
Indeed, even moved it was not triggered. I have the feeling that `symlinkJoin` does not trigger automatically hooks from `nativeBuildInputs`. Am I correct ?

Reply:
Haven't looked into it but that assumption sounds reasonable.